Assignment/Conversion.c

```diff
@@ -0,0 +1,87 @@
+// ------------------------------------------------------------------------------------------------
+// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.
+// ------------------------------------------------------------------
+// Conversion.c
+// Program for Number Conversion.
+// ------------------------------------------------------------------------------------------------
+
+#pragma warning(disable:4996)
+#include <stdio.h>
+
+void DecToBinary (int n);
+void DecToHexa (int n);
+int mBitLength;
+/// <summary>Decimal number to binary Conversion</summary>
+void DecToBinary (int n) {
+   int binary[32], i = 0, isNegative = 0; // Flag is to check if the number is negative
+   if (n == 0) {
+      printf ("0");
+      return;
+   }
+   if (n < 0) {
+      n = -n;
+      isNegative = 1;
+   }
+   while (n > 0) {
+      binary[i++] = n % 2;
+      n /= 2;
+   }
+   // If the number was negative,apply two's complement
+   if (isNegative) {

```

Comment:
It's really nice you've coded the whole logic of 2's complement, but you may want to look at the [bit shift operators (>> & <<)](https://stackoverflow.com/questions/141525/what-are-bitwise-shift-bit-shift-operators-and-how-do-they-work) and see how to improve your code using that.